### PR TITLE
Cleanups

### PR DESCRIPTION
**Pull request by djc**
My goal here is to make the codebase easier to navigate and take advantage of the MSRV upgrade from #653 to employ more modern Rust standards.

### CODE REVIEW

File: src/datetime/rustc_serialize.rs [hunk 1]
```diff
@@ -0,0 +1,119 @@
+use super::DateTime;
+#[cfg(feature = "clock")]
+use crate::offset::Local;
+use crate::offset::{FixedOffset, LocalResult, TimeZone, Utc};
+use core::fmt;
+use core::ops::Deref;
+use rustc_serialize::{Decodable, Decoder, Encodable, Encoder};
```
Comment (Milo123459) [inline src/datetime/rustc_serialize.rs:7]: Why use rustc_serialize? Serde?

**Review by Milo123459 — APPROVED** [review]
LGTM, although serde should be the default.

File: src/datetime/rustc_serialize.rs (hunk 1 above)
Comment (djc) [reply src/datetime/rustc_serialize.rs:7]: I'm not making any backwards incompatible changes here, so that means keeping rustc_serialize.

**Comment by djc** [comment]
> LGTM, although serde should be the default.

Do you mean that we should enable the serde feature by default? I don't agree -- most well-maintained libraries keep it as an off-by-default feature, so that downstream users who don't need it don't have to compile the extra dependency.

**Pull request merged**